addons/mail/models/mail_mail.py

```diff
@@ -481,6 +484,31 @@ def _send(self, auto_commit=False, raise_exception=False, smtp_session=None):
                     except Exception:
                         pass
 
+                # Compute email size and transform attachment in downloadable link in the body if it exceeds the limit
+                estimated_email_size = (
+                        (len(bytes(json.dumps(headers), "utf-8")) +
```

Comment:
I don't really get why you need to convert to byte to compute the length, it should be (almost) always the same isn't it ?
```suggestion
                        (len(json.dumps(headers)) +
```

Reply:
Good catch, json.dumps encodes everything by default in ascii making a non ascii text very long; each non ascii characters takes 5 times more place! (instead of 2 times most of the time, see example below).

So the idea is to set ensure_ascii=False to have a better estimation. After the dump we still need to encode it in UTF-8 in order to count the bytes and not the characters. Indeed the goal is to evaluate the message size in byte and not in characters. For example, "é" will takes 2 bytes.

Here we assume that the worst/common encoding is to use utf-8 + base64.

Note that bytes('é', 'utf-8') is the same as 'é'.encode(). But indeed ".encode()" is easier so I have replaced it.

Detailed example: 

{'k': 'é'} --> json.dumps(text) --> {"k": "\u00e9"} --> len=15
        --> print(len(json.dumps(text).encode(errors='ignore'))) = 15
        
{'k': 'é'} --> json.dumps(text, ensure_ascii=False) --> {"k": "é"} --> len=10
        --> len(json.dumps(text, ensure_ascii=False).encode(errors='ignore')) = 11